Migration to modern gazebo

AdarshKaran opened the pull request:
## MP_400 Updates

Changes & Improvements
- Migration to modern Gazebo
- Refactored the `urdf` and folder structure
- Updated `navigation.yaml` to support migration
- Modified `bringup.launch.py` by adding `scanner_type` argument
- Fully refactored `navigation.launch.py` using opaque function
- Updated maps for using `neo_gz_worlds` package
- Tested simulation and navigation
- Removed `rviz.launch.py` → this will be used from the new `mp_rviz` package

AdarshKaran committed:
Completed migration to modern gazebo, reafactore launch files, modularized and restructured urdf

padhupradheep commented on configs/navigation/navigation.yaml:
```diff
@@ -291,11 +292,17 @@ global_costmap:
         cost_scaling_factor: 4.5
         inflation_radius: 0.8
       always_send_full_costmap: True
+  global_costmap_client:
```
This is not needed

padhupradheep commented on configs/navigation/navigation.yaml:
```diff
@@ -291,11 +292,17 @@ global_costmap:
+  global_costmap_client:
+    ros__parameters:
+      use_sim_time: False
+  global_costmap_rclcpp_node:
```
This too

padhupradheep commented on launch/navigation.launch.py:
```diff
@@ -69,32 +59,17 @@ def generate_launch_description():
 
         IncludeLaunchDescription(
             PythonLaunchDescriptionSource([nav2_launch_file_dir, '/navigation_neo.launch.py']),
-            condition=IfCondition(PythonExpression(['not ', docking])),
```
revert this please

padhupradheep commented on launch/navigation.launch.py:
```diff
@@ -69,32 +59,17 @@ def generate_launch_description():
-
-        IncludeLaunchDescription(
-            PythonLaunchDescriptionSource([nav2_launch_file_dir, '/navigation_neo.launch.py']),
-            condition=IfCondition(docking),
```
this too, we need docking separately

AdarshKaran replied:
I have added a logic to conditionally change the params file based on the enable_docking launch argument now

padhupradheep commented on launch/navigation.launch.py:
```diff
@@ -117,12 +92,95 @@ def generate_launch_description():
             output='screen',
             parameters=[{'use_sim_time': use_sim_time},
-                        {'autostart': autostart},
+						{'autostart': autostart},
```
why tab?

padhupradheep commented on robot_model/xacros/mp_400_wheel_macro.xacro:
```diff
@@ -0,0 +1,59 @@
+</robot>
+
+
+   
```
why is there a `-` EoF is somehow screwed up

padhupradheep commented on robot_model/xacros/sick_microscan3_macro.xacro:
```diff
@@ -0,0 +1,50 @@
+    </link>
+
+  </xacro:macro>
+</robot>
```
Check EoF

padhupradheep commented on robot_model/xacros/uss.xacro:
```diff
@@ -13,9 +18,17 @@
+
   </xacro:macro>
   
 </robot>
```
update EoF here as well

padhupradheep commented on robot_model/xacros/sick_s300_macro.xacro:
```diff
@@ -0,0 +1,50 @@
+    </link>
+
+  </xacro:macro>
+</robot>
```
Fix EoF here too

padhupradheep commented on robot_model/xacros/mp_400_caster_wheel_macro.xacro:
```diff
@@ -0,0 +1,52 @@
+      
+  </xacro:macro>
+
+</robot>
```
EoF check

padhupradheep commented on configs/navigation/navigation.yaml:
Did you check if navigation works on both rolling and jazzy?

AdarshKaran replied:
I have tested the navigation and monitored the odom / cmd_vel values too for rolling. Once these changes are finalized I will update it for the rest of the mp series robots and test them all in jazzy

padhupradheep reviewed (changes requested):
Good work! 

I still wonder:

- [x] where to find the GZ-ROS bridge?
- [x] if navigation works on rolling and jazzy
- [x] if we should have a bringup-sim.launch.py. I understand we have that in mp_utils, but would be more convenient for the users, if we can provide the tools to launch it like `ros2 launch neo_mp_400-2 bringup_sim.launch.py`

AdarshKaran committed:
Addressing PR #33 comments, updated navigation.yaml, adding docking condition, fixed indentation and formatting issues, updated dependencies

AdarshKaran commented:
> Good work!
> 
> I still wonder:
> 
>     * [ ]  where to find the GZ-ROS bridge?
> 
>     * [ ]  if navigation works on rolling and jazzy
> 
>     * [ ]  if we should have a bringup-sim.launch.py. I understand we have that in mp_utils, but would be more convenient for the users, if we can provide the tools to launch it like `ros2 launch neo_mp_400-2 bringup_sim.launch.py`

1. the gz-ros2 bridge yaml is located in `mp_utils/mp_bringup/configs/gz_bridge/gz_bridge_config.yaml`  
2. I have tested the navigation in rolling and it works fine. once these changes are finalized I will test it in jazzy also.
3. I have added a `bringup_sim.launch.py` file that accepts launch args and uses the `bringup_sim.launch.py` from the mp_bringup package.

AdarshKaran commented:
Changes & Improvements
- Added `bringup_sim.launch.py`
- Removed additional global costmap parameters in `navigation.yaml`  
- Added `mp_utils` dependencies to `mp` robot `package.xml`  
- `mp-400` launch files -> fixed indentations, removed additional tabs and added empty line at the end  
- Added docking params `navigation.yaml` and docking condition in `navigation.launch` which chooses the appropriate file based on `enable_docking` launch argument  
- Removed `use_sim_time` parameters from `nav2` params files

padhupradheep commented:
Very Good Adarsh

padhupradheep reviewed (approved):
Good job
